### PR TITLE
upgrade rust to 1.68.0, address some vendoring issues and promote libgit2 to core

### PR DESCRIPTION
**Pull request by nicogbg**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

upgrade rust to 1.68.0, address some vendoring issues and promote libgit2 to core

###### Change Log  

- Change
- Change
- Change

###### Does this affect the toolchain?  

**NO**

###### Associated issues  

- #xxxx

###### Links to CVEs  
- https://nvd.nist.gov/vuln/detail/CVE-YYYY-XXXX

###### Test Methodology

- Pipeline build id: xxxx

### CODE REVIEW

File: SPECS/rust/rust.spec [hunk 1]
```diff
@@ -28,18 +32,29 @@ Source6:        https://static.rust-lang.org/dist/%{release_date}/rustc-%{stage0
 Source7:        https://static.rust-lang.org/dist/%{release_date}/rust-std-%{stage0_version}-aarch64-unknown-linux-gnu.tar.gz
 BuildRequires:  binutils
 BuildRequires:  cmake
+# make sure rust relies on curl from CBL-Mariner (instead of using its vendored flavor)
 BuildRequires:  curl-devel
 BuildRequires:  git
 BuildRequires:  glibc
+# make sure rust relies on libgit2 from CBL-Mariner (instead of using its vendored flavor)
+BuildRequires:  libgit2-devel
```
Comment (mfrw) [inline SPECS/rust/rust.spec:40]: Question: How would this ensure that we are using this version of `libgit2` and not the one that is in our cache ?

Comment (nicogbg) [reply SPECS/rust/rust.spec:40]: this is by design in rust code (see https://github.com/rust-lang/git2-rs/blob/master/libgit2-sys/build.rs)

Comment (nicogbg) [reply SPECS/rust/rust.spec:40]: I also verified after installing rust that cargo uses our libgit2 (and others) using 'ldd cargo...')

File: SPECS/libgit2/libgit2.spec [hunk 2]
```diff
@@ -1,7 +1,7 @@
 Summary:        C implementation of the Git core methods as a library with a solid API
 Name:           libgit2
 Version:        1.4.5
-Release:        1%{?dist}
```
Comment (jslobodzian) [inline SPECS/libgit2/libgit2.spec:4]: Please double check CVE’s against this package before submitting

**Review by jslobodzian — APPROVED** [review]
Approved, but please check the libgit2 package for any CVE’s first

**Pull request merged**